.github/workflows/testing.yml

```diff
@@ -23,25 +21,14 @@ jobs:
         uses: actions/cache@v3
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements-dev.txt') }}
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('tox.ini') }}
```

Comment:
The hash still needs to include requirements-dev.txt and/or .pre-commit-config.yaml in order to clear the cache when the requirements are updated.